.gitlab/lassen-jobs.yml

```diff
@@ -75,18 +75,18 @@ gcc_8_3_1_cuda:
 
 pgi_19_10_cuda:
   variables:
-    SPEC: "+fortran+cuda %pgi@19.10 ^cuda@10.1.168"
+    SPEC: "+fortran+cuda %pgi@19.10 cxxflags=-w ^cuda@10.1.168"
   extends: .build_and_test_on_lassen
 
 pgi_19_10_gcc_8_3_1_cuda:
   variables:
-    SPEC: "+fortran+cuda %pgi@19.10 cxxflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 cflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 ^cuda@10.1.168"
+    SPEC: "+fortran+cuda %pgi@19.10 cxxflags='--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 -w' cflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 ^cuda@10.1.168"
   allow_failure: true
   extends: .build_and_test_on_lassen
 
 pgi_20_4_gcc_8_3_1_cuda_11:
   variables:
-    SPEC: "+fortran+cuda %pgi@20.4 cxxflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 cflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 ^cuda@11.0.2"
+    SPEC: "+fortran+cuda %pgi@20.4 cxxflags='--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 -w' cflags=--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.3.1 ^cuda@11.0.2"
```

Comment:
What do these changes have to do with updating the tutorial examples?  They seem more general in nature.  I suggest that these be broken out to their own PR.

Reply:
This PR includes the deprecation. This is to avoid warnings from that, that will become errors during CI builds. These extra flags should be handled in `cmake/SetupCompilerFlags.cmake` like they are for the other compilers.